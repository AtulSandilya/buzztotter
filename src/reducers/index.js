import { combineReducers } from 'redux';

import { settings } from './settings';
import { view } from './view';

export default combineReducers({
  settings,
  view,
})
