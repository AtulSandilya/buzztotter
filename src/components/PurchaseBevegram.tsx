import * as React from "react";
import { Component, PropTypes } from 'react';
import {
  ActivityIndicator,
  Alert,
  Image,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View
} from 'react-native';

import Ionicon from 'react-native-vector-icons/Ionicons';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

import {isIOS, WindowWidth} from '../Utilities';

import { CreditCard, PurchaseData, PurchaseState, PurchasePackage} from '../reducers/purchase';

import {SendBevegramData} from '../sagas/sendBevegram';
import {InProgressData} from './PurchaseAndOrSendInProgress';

import StatusLine from './StatusLine';
import RouteWithNavBarWrapper from './RouteWithNavBarWrapper';
import TitleText from './TitleText';
import BevButton, {getButtonHeight} from './BevButton';

import {globalColors, globalStyles, BevLayoutAnimation} from './GlobalStyles';

export const FormatCreditCardBrandForFontAwesomeIcon = (card: CreditCard) => {
  const cardMap = {
    "Visa": "visa",
    "MasterCard": "mastercard",
    "American Express": "amex",
    "Discover": "discover",
    "JCB": "jcb",
    "Diners Club": "diners-club",
  }

  const cardPrefix = "cc-"

  if(card && cardMap[card.brand]){
    return cardPrefix + cardMap[card.brand];
  }

  return "credit-card";
}

interface PurchaseBevegramProps {
  userBevegrams: number;
  fullName: string;
  firstName: string;
  imageUri: string;
  facebookId: string;
  purchase: PurchaseState;
  creditCards: CreditCard[];
  activeCardId: string;
  attemptingUpdate: boolean;
  attemptingVerification: boolean;
  purchasePackages: PurchasePackage[];
  selectedPurchasePackage: PurchasePackage;
  selectedPurchasePackageIndex: number;
  attemptingSend: boolean;
  completedSend: boolean;
  resetPurchase();
  closePurchaseRoute();
  goToAddCreditCardRoute();
  removeCard(id, index): void;
  updateDefaultCard(newDefaultCardId): void;
  selectPackage(packageId): void;
  startCreditCardPurchase(PurchaseData, InProgressData): void;
  sendBevegram(SendBevegramData, InProgressData): void;
  purchaseAndSend(PurchaseData, SendBevegramData, InProgressData): void;
}

interface PurchaseBevegramState {
  promoCode: string;
  message: string;
  bevegramsToSend: number;
  // Track the number of userBevegrams when this component is rendered. Allows
  // userBevegrams to change (which it does after a purchase and a send) but
  // base the view logic on the initial userBevegram count.
  bevegramsBeforePurchaseOrSendOrBoth: number;
}

export default class PurchaseBevegram extends Component<PurchaseBevegramProps, PurchaseBevegramState> {

  constructor(props){
    super(props);
    this.state = {
      promoCode: "",
      message: "",
      bevegramsToSend: 1,
      bevegramsBeforePurchaseOrSendOrBoth: this.props.userBevegrams,
    };
  }

  buttonFontSize = 20;

  componentWillUpdate() {
    BevLayoutAnimation();
  }

  userIsSending(): boolean {
    return this.props.fullName !== undefined;
  }

  userIsPurchasing(): boolean {
    if(!this.userIsSending()) return true;
    return (this.state.bevegramsToSend > this.state.bevegramsBeforePurchaseOrSendOrBoth);
  }

  userIsPurchasingAndSending(): boolean {
    return this.userIsPurchasing() && this.userIsSending();
  }

  updateBevegramsToSend(amount){
    const min = 1;
    const userBevs = this.props.userBevegrams;
    const packagesMax = this.props.purchasePackages.slice(-1)[0].quantity;
    const max = packagesMax + userBevs;
    const newAmount = this.state.bevegramsToSend + amount;
    const bevsToSend = this.state.bevegramsToSend;
    const packageIndex = this.props.selectedPurchasePackageIndex;
    const packages = this.props.purchasePackages;

    if(newAmount <= max && newAmount >= min){
      this.updateState("bevegramsToSend", newAmount);


      // Automatically select the purchasePackage that is equal to or above
      // the users Bevegrams
      const bevsUserMustPurchase = (bevsToSend + amount) - userBevs;
      if(bevsToSend > userBevs){
        if(amount > 0){
          if(packageIndex + 1 !== packages.length
            && bevsUserMustPurchase > packages[packageIndex].quantity){
            this.props.selectPackage(packageIndex + 1);
          }
        }

        if(amount < 0){
          if(packageIndex > 0
            && (bevsUserMustPurchase <= packages[packageIndex - 1].quantity)){
            this.props.selectPackage(packageIndex - 1);
          }
        }
      }
    }
  }

  increaseBevegramsToSend(){
    this.updateBevegramsToSend(1);
  }

  decreaseBevegramsToSend(){
    this.updateBevegramsToSend(-1);
  }

  updateState(property, value){
    this.setState(function(prevState, currentProps){
      let nextState = Object.assign({}, prevState);
      nextState[property] = value;
      return nextState;
    });
  }

  onSelectPackage(newSelectedPurchasePackageIndex){
    // Warn the user if the package they are choosing is smaller than the
    // number of bevegrams they want to send
    const newPack = this.props.purchasePackages[newSelectedPurchasePackageIndex];
    if(newPack.quantity < (this.props.userBevegrams + this.state.bevegramsToSend)){
      alert("This package is too small!");
      return;
    }

    this.props.selectPackage(newSelectedPurchasePackageIndex);
  }

  initiatePurchaseOrSendOrBoth() {
    if(this.userIsPurchasing()){
      if(this.props.creditCards.length === 0){
        alert("Please Add a Credit Card!");
        return;
      }
    }

    if(this.userIsPurchasingAndSending()){
      this.props.purchaseAndSend(this.packPurchaseData(), this.packSendData(), this.packInProgressData());
    } else if(this.userIsPurchasing()){
      this.props.startCreditCardPurchase(this.packPurchaseData(), this.packInProgressData());
    } else if(this.userIsSending()){
      this.props.sendBevegram(this.packSendData(), this.packInProgressData());
    }
  }

  packPurchaseData(): PurchaseData {
    const pack = this.props.selectedPurchasePackage;
    return {
      amount: pack.price * 100,
      description: `Purchased ${pack.quantity} Bevegram${pack.quantity > 1 ? "s" : ""} via Buzz Otter`,
      quantity: pack.quantity,
    }
  }

  packSendData(): SendBevegramData {
    return {
      recipentName: this.props.fullName,
      facebookId: this.props.facebookId,
      quantity: this.state.bevegramsToSend,
      message: this.state.message,
    }
  }

  packInProgressData(): InProgressData {
    const activeCard = this.getActiveCard();
    return {
      recipentImage: this.props.imageUri,
      bevegramsUserIsSending: this.state.bevegramsToSend,
      bevegramsUserIsPurchasing: this.packPurchaseData().quantity,
      bevegramsPurchasePrice: this.props.selectedPurchasePackage.price.toFixed(2),
      cardLast4: activeCard.last4,
      cardFontAwesomeIcon: FormatCreditCardBrandForFontAwesomeIcon(activeCard),
      userIsPurchasing: this.userIsPurchasing(),
      userIsSending: this.userIsSending(),
      recipentFullName: this.props.fullName,
      buttonFontSize: this.buttonFontSize,
    }
  }

  getActiveCard(){
    // Is this a bad idea?
    let activeCard: CreditCard = {
      token: undefined,
      brand: undefined,
      last4: undefined,
      id: undefined,
    };

    this.props.creditCards.map((card) => {
      if(card.id === this.props.activeCardId){
        activeCard = card;
      }
    });

    return activeCard;
  }

  renderSendOptions() {
    if(!this.userIsSending()){
      return null;
    }

    return (
      <View style={{flex: 1}}>
        <View style={globalStyles.bevLine}>
          <View style={globalStyles.bevLineLeft}>
            <Text style={globalStyles.bevLineTextTitle}>Receipent:</Text>
          </View>
          <View style={globalStyles.bevLineRight}>
            <Image
              source={{uri: this.props.imageUri}}
              style={{
                height: 40,
                width: 40,
                marginRight: 10,
              }}
            />
            <Text style={globalStyles.bevLineText}>{this.props.fullName}</Text>
          </View>
        </View>
        <View style={globalStyles.bevLine}>
          <View style={globalStyles.bevLineLeft}>
            <Text style={globalStyles.bevLineTextTitle}>Bevegrams:</Text>
          </View>
          <View style={globalStyles.bevLineRight}>
            <View style={{
              flex: 1,
              flexDirection: 'row',
              alignItems: 'center',
              justifyContent: 'flex-end',
            }}>
              <Text style={globalStyles.bevLineTextTitle}>{this.state.bevegramsToSend}</Text>
              <TouchableHighlight
                underlayColor={"transparent"}
                onPress={() => this.increaseBevegramsToSend()}
                hitSlop={{top: 10, left: 10, bottom: 10, right: 10}}
                style={{marginLeft: 15}}>
                  <FontAwesome
                    name="plus-circle"
                    style={globalStyles.bevIcon}
                    color="#555555"
                    size={28}
                  />
              </TouchableHighlight>
              <TouchableHighlight
                underlayColor={"transparent"}
                onPress={() => this.decreaseBevegramsToSend()}
                hitSlop={{top: 10, left: 10, bottom: 10, right: 10}}
                style={{marginLeft: 15}}>
                <FontAwesome
                  name="minus-circle"
                  style={globalStyles.bevIcon}
                  color="#555555"
                  size={28}
                />
              </TouchableHighlight>
            </View>
          </View>
        </View>
        <View style={globalStyles.bevLineNoSep}>
          <Text style={globalStyles.bevLineTextTitle}>Message:</Text>
          <Text style={globalStyles.bevLineText}> (Optional)</Text>
        </View>
        <View style={globalStyles.bevLine}>
          <TextInput
            placeholder={"Happy Birthday! Have a cold one on me!"}
            placeholderTextColor={"#cccccc"}
            style={{
              flex: 1,
              height: 45,
            }}
            onChangeText={(text) => {
              this.updateState("message", text);
            }}
          />
        </View>
      </View>
    )
  }

  renderPurchaseOptions() {
    if(!this.userIsPurchasing()){
      return null;
    }

    return (
      <View style={{flex: 1}}>
        <View style={globalStyles.bevLineNoSepWithMargin}>
          <View style={{flex: 1, flexDirection: 'row'}}>
            <View style={globalStyles.bevLineLeft}>
              <Text style={globalStyles.bevLineTextTitle}>
                Purchase Bevegrams:
              </Text>
            </View>
          </View>
        </View>
        {this.props.purchasePackages.map((pack, index) => {
          return (
            <TouchableHighlight
              underlayColor={"transparent"}
              style={globalStyles.bevLine}
              key={"package" + index}
              onPress={this.onSelectPackage.bind(this, index)}
            >
              <View style={{flex: 1, flexDirection: 'row'}}>
                <View style={[globalStyles.bevLineLeft, {flexDirection: 'row'}]}>
                  {(this.props.selectedPurchasePackageIndex === index) && this.userIsPurchasing() ?
                    <FontAwesome
                      name="check-square-o"
                      color="green"
                      size={25}
                      style={globalStyles.bevIcon}
                    />
                  :
                    <FontAwesome
                      name="square-o"
                      size={25}
                      color="#999"
                      style={globalStyles.bevIcon}
                    />
                  }
                  <Text style={globalStyles.bevLineTextTitle}>
                    {pack.name}
                  </Text>
                </View>
                <View style={globalStyles.bevLineRight}>
                  <Text style={globalStyles.bevLineText}>$ {pack.price.toFixed(2)}</Text>
                </View>
              </View>
            </TouchableHighlight>
          )
        })}
        <View style={globalStyles.bevLine}>
          <View style={globalStyles.bevLineLeft}>
            <Text style={globalStyles.bevLineTextTitle}>Promo Code:</Text>
          </View>
          <View style={globalStyles.bevLineRight}>
            <TextInput
              style={{
                width: 65,
                textAlign: 'center',
                height: 40,
              }}
              ref="promoCodeInput"
              maxLength={4}
              placeholder={"ABCD"}
              placeholderTextColor={"#cccccc"}
              onChangeText={(text) => {
                this.updateState("promoCode", text);
              }}
            />
          </View>
        </View>
        {this.props.creditCards ? this.props.creditCards.map((card, index) => {
          return (
            <View style={globalStyles.bevLine} key={card.id}>
              <TouchableHighlight
                underlayColor={"transparent"}
                onPress={() => {
                  if(!this.props.attemptingUpdate
                     && this.props.creditCards.length > 1
                     && card.id !== this.getActiveCard().id
                    ){
                    this.props.updateDefaultCard(card.id);
                  }
                }}
                style={[globalStyles.bevLineLeft, {
                  flexDirection: 'row',
                  alignItems: 'flex-start',
                  justifyContent: 'flex-start',
              }]}>
                <View style={{flex: -1, flexDirection: 'row'}}>
                  <View
                    style={{
                    paddingRight: 15,
                    alignItems: 'center',
                    justifyContent: 'center',
                    flex: -1,
                    marginTop: 4,
                  }}>
                    {this.props.attemptingUpdate ?
                        <ActivityIndicator style={{height: 28, width: 28}} />
                    :
                      card.id === this.getActiveCard().id ?
                        <FontAwesome
                          name="check-square-o"
                          size={25}
                          color="green"
                          style={globalStyles.bevIcon}
                        />
                      :
                        <FontAwesome
                          name="square-o"
                          size={25}
                          color="#999"
                          style={globalStyles.bevIcon}
                        />
                    }
                  </View>
                  <View style={{flex: -1, flexDirection: "row", justifyContent: 'center', alignItems: 'center'}}>
                    <FontAwesome name={"cc-" + card.brand.toLowerCase()} size={30} style={{paddingRight: 10}}/>
                    <Text style={globalStyles.bevLineText}>.... {card.last4}</Text>
                  </View>
                </View>
              </TouchableHighlight>
              <View style={globalStyles.bevLineRight}>
                <TouchableHighlight
                underlayColor={"transparent"}
                  onPress={() => {
                    if(!this.props.attemptingUpdate){
                      this.props.removeCard(card.id, index);
                    }
                  }}
                  style={{
                    paddingRight: 10,
                  }}>
                  <Text style={{color: '#999'}}>{this.props.attemptingUpdate ? "Updating..." : "Remove"}</Text>
                </TouchableHighlight>
              </View>
            </View>
          )
        }) : <View/>}
        <View style={[globalStyles.bevLineNoSepWithMargin]}>
          <TouchableHighlight
            underlayColor={"transparent"}
            style={{
              flex: 1,
              flexDirection: 'row',
            }}
            onPress={() => {
              this.props.goToAddCreditCardRoute();
            }}
          >
            <View style={{flex: 1, flexDirection: 'row'}}>
              <View style={[globalStyles.bevLineLeft, {flex: 2}]}>
                <Text
                  style={[globalStyles.bevLineTextTitle, {
                    fontWeight: "normal"
                  }]}>
                  {this.props.attemptingVerification ?
                    "Adding Credit Card..."
                    :
                    "Add Credit Card"
                  }
                </Text>
              </View>
              <View style={globalStyles.bevLineRight}>
                <Ionicon name="ios-arrow-forward" size={35} />
              </View>
            </View>
          </TouchableHighlight>
        </View>
      </View>
    )
  }

  renderSendAndPurchaseOptions(){
    const listFlex = 1;
    const bottomButtonFlex = -1;
    let buttonText: string;
    const purchaseText = "Purchase";
    const sendText = "Send";
    if(this.userIsPurchasingAndSending()){
      buttonText = `${purchaseText} & ${sendText}`
    } else if (this.userIsPurchasing()){
      buttonText = purchaseText;
    } else if (this.userIsSending()){
      buttonText = sendText;
    }
    const purchaseButtonIcon = this.userIsPurchasing() ?
      FormatCreditCardBrandForFontAwesomeIcon(this.getActiveCard())
      : "paper-plane";
    const viewBelowHeight = getButtonHeight(this.buttonFontSize);

    return(
      <View style={{flex: 1}}>
        <View style={{flex: listFlex, backgroundColor: "#ffffff"}}>
          <RouteWithNavBarWrapper
            viewBelowHeight={viewBelowHeight}
          >
            <View style={[globalStyles.bevContainer]}>
              {this.renderSendOptions()}
              {this.renderPurchaseOptions()}
              {/* Add empty view to ensure elements above are viewable */}
              <View style={{height: viewBelowHeight + 20, width: WindowWidth}}></View>
            </View>
          </RouteWithNavBarWrapper>
        </View>
        <View style={{
          flex: bottomButtonFlex,
          shadowColor: '#333333',
          shadowOpacity: 0.15,
          shadowRadius: 2,
          elevation: 15,
          zIndex: 1,
          backgroundColor: '#ffffff',
        }}>
          <View style={[globalStyles.bevContainer, {flex: -1, margin: 0, paddingVertical: 10, paddingHorizontal: 10}]}>
            <View style={[globalStyles.bevLineNoSep, {height: getButtonHeight(this.buttonFontSize), paddingBottom: 0, alignItems: 'center', justifyContent: 'center'}]}>
              <View style={[globalStyles.bevLineLeft]}>
                <BevButton
                  onPress={this.props.closePurchaseRoute}
                  text={this.userIsPurchasingAndSending() ? "" : "Cancel"}
                  shortText={""}
                  fontAwesomeLeftIcon="ban"
                  label="Cancel Purchase Button"
                  buttonFontSize={this.buttonFontSize}
                  margin={0}
                />
              </View>
              <View style={globalStyles.bevLineRight}>
                <BevButton
                  onPress={this.initiatePurchaseOrSendOrBoth.bind(this)}
                  text={buttonText}
                  shortText={buttonText}
                  label={buttonText + " Button"}
                  buttonFontSize={this.buttonFontSize}
                  fontAwesomeLeftIcon={purchaseButtonIcon}
                  margin={0}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    );
  }

  render() {
    return this.renderSendAndPurchaseOptions();
  }
}

interface Style {
  numBeersContainer: React.ViewStyle;
  numBeersButtonContainer: React.ViewStyle;
  numBeersButton: React.ViewStyle;
  numBeersButtonText: React.TextStyle;
}

const styles = StyleSheet.create<Style>({
  numBeersContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  numBeersButtonContainer: {
    flex: -1,
    flexDirection: 'row',
  },
  numBeersButton: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: globalColors.bevPrimary,
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 100,
  },
  numBeersButtonText: {
    fontSize: 25,
    fontWeight: 'bold',
  },
});
