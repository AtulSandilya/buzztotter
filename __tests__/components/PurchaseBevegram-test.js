import 'react-native';
import React from 'react';
import PurchaseBevegram from '../../build/components/PurchaseBevegram';

import renderer from 'react-test-renderer';

import { initialPurchaseState } from '../../build/reducers/purchase';

it('renders correctly', () => {
  const purchasePackages = [{
      name: "One",
      quantity: 1,
      price: 7.50,
  }]
  const tree = renderer.create(
    <PurchaseBevegram
      fullName="Testy McTest"
      firstName="Testy"
      purchase={initialPurchaseState}
      creditCards={[]}
      selectedPurchasePackage={purchasePackages[0]}
      purchasePackages={purchasePackages}
      imageUri="test.jpg"
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
