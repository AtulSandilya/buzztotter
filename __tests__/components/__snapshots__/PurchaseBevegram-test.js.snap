exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }>
    <ScrollView
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutoAutomaticScroll={true}
      extraHeight={75}
      extraScrollHeight={0}
      keyboardDismissMode="interactive"
      onScroll={[Function]}
      scrollEventThrottle={8}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "flex": 1,
          "height": 1197,
          "left": 0,
          "top": 85,
          "width": 750,
        }
      }>
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "paddingBottom": 5,
              "paddingHorizontal": 20,
              "paddingTop": 15,
            },
          ]
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#dddddd",
                "flex": -1,
                "flexDirection": "row",
                "marginBottom": 15,
                "paddingBottom": 10,
              }
            }>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": -1,
                  "justifyContent": "center",
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }>
                Receipent:
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }>
              <Image
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "height": 40,
                    "marginRight": 10,
                    "width": 40,
                  }
                } />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 20,
                  }
                }>
                Testy McTest
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#dddddd",
                  "flex": -1,
                  "flexDirection": "row",
                  "marginBottom": 15,
                  "paddingBottom": 10,
                },
              ]
            }
            testID={undefined}>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-start",
                      "flex": -1,
                      "justifyContent": "center",
                    },
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#999",
                        "fontSize": 25,
                      },
                      Object {
                        "flex": -1,
                        "height": 28,
                        "width": 28,
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }>
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                      "fontWeight": "bold",
                    }
                  }>
                  One
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }>
                  $ 
                  7.50
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#dddddd",
                "flex": -1,
                "flexDirection": "row",
                "marginBottom": 15,
                "paddingBottom": 10,
              }
            }>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": -1,
                  "justifyContent": "center",
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }>
                Promo Code:
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }>
              <TextInput
                maxLength={4}
                onChangeText={[Function]}
                placeholder="ABCD"
                placeholderTextColor="#cccccc"
                style={
                  Object {
                    "height": 40,
                    "textAlign": "center",
                    "width": 65,
                  }
                } />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#dddddd",
                  "flex": -1,
                  "flexDirection": "row",
                  "marginBottom": 15,
                  "paddingBottom": 10,
                },
              ]
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  },
                ]
              }
              testID={undefined}>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "flex-start",
                        "flex": -1,
                        "justifyContent": "center",
                      },
                      Object {
                        "flex": 2,
                      },
                    ]
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                        },
                        Object {
                          "fontWeight": "normal",
                        },
                      ]
                    }>
                    Add Credit Card
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "flex-end",
                    }
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 35,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }>
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <View
            style={
              Object {
                "flex": -1,
                "flexDirection": "row",
                "paddingBottom": 10,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }>
              Message:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                }
              }>
               (Optional)
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#dddddd",
                "flex": -1,
                "flexDirection": "row",
                "marginBottom": 15,
                "paddingBottom": 10,
              }
            }>
            <TextInput
              onChangeText={[Function]}
              placeholder="Happy Birthday! Have a cold one on me!"
              placeholderTextColor="#cccccc"
              style={
                Object {
                  "flex": 1,
                  "height": 45,
                }
              } />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 72,
              "width": 750,
            }
          } />
      </View>
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 15,
        "flex": -1,
        "shadowColor": "#333333",
        "shadowOpacity": 0.15,
        "shadowRadius": 2,
        "zIndex": 1,
      }
    }>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "paddingBottom": 5,
            "paddingHorizontal": 20,
            "paddingTop": 15,
          },
          Object {
            "flex": -1,
            "margin": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          },
        ]
      }>
      <View
        style={
          Array [
            Object {
              "flex": -1,
              "flexDirection": "row",
              "paddingBottom": 10,
            },
            Object {
              "alignItems": "center",
              "height": 52,
              "justifyContent": "center",
              "paddingBottom": 0,
            },
          ]
        }>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": -1,
                "justifyContent": "center",
              },
            ]
          }>
          <View
            accessibilityLabel="Cancel Purchase Button"
            style={
              Object {
                "flex": -1,
                "flexDirection": "row-reverse",
              }
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#8B5E3C",
                      "borderColor": "#000000",
                      "borderRadius": 10,
                      "elevation": 5,
                      "flex": -1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "margin": 15,
                      "padding": 15,
                      "shadowColor": "#333333",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 2,
                      },
                      "shadowOpacity": 0.5,
                      "shadowRadius": 0.95,
                    },
                    Object {
                      "margin": 0,
                    },
                    null,
                    null,
                    null,
                    Object {
                      "height": 52,
                    },
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgba(0, 0, 0, 0)",
                          "color": "#ffffff",
                        },
                        Object {
                          "fontSize": 30,
                          "paddingTop": 2,
                        },
                        Object {
                          "paddingRight": 0,
                        },
                      ],
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }>
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }>
          <View
            accessibilityLabel="Purchase & Send Button"
            style={
              Object {
                "flex": -1,
                "flexDirection": "row-reverse",
              }
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#8B5E3C",
                      "borderColor": "#000000",
                      "borderRadius": 10,
                      "elevation": 5,
                      "flex": -1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "margin": 15,
                      "padding": 15,
                      "shadowColor": "#333333",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 2,
                      },
                      "shadowOpacity": 0.5,
                      "shadowRadius": 0.95,
                    },
                    Object {
                      "margin": 0,
                    },
                    null,
                    null,
                    null,
                    Object {
                      "height": 52,
                    },
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgba(0, 0, 0, 0)",
                          "color": "#ffffff",
                        },
                        Object {
                          "fontSize": 30,
                          "paddingTop": 2,
                        },
                        Object {
                          "paddingRight": 10,
                        },
                      ],
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }>
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0)",
                        "color": "#ffffff",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ]
                  }>
                  Purchase & Send
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
